packages/react-app/src/contexts/SettingsContext.jsx

```diff
@@ -15,6 +16,11 @@ export const SettingsProvider = ({ children }) => {
   const [xdaiRPC, setXDaiRPC] = useState('');
   const [neverShowClaims, setNeverShowClaims] = useState(false);
   const [infiniteUnlock, setInfiniteUnlock] = useState(false);
+  const [disableBalanceFetchToken, setDisableBalanceFetchToken] = useLocalState(
+    false,
+    'disable-balance-while-token-fetch',
```

Comment:
This refactor is great! Can we use the same for all the other settings?
Also  please store this string in constants and pick it up from there :)